grpchantesting/test_service.go

```diff
@@ -15,7 +15,9 @@ import (
 )
 
 // TestServer has default responses to the various kinds of methods.
-type TestServer struct{}
+type TestServer struct {
+	UnimplementedTestServiceServer
```

Comment:
Ick, this is a requirement of the latest gRPC generator (`protoc-gen-go-grpc`).

Luckily, it's actually possible to _disable_ the requirement, which is what I figure we'll do when it comes time to update our private repos. For this one, it was a simple one line to make the generated code happy, so I figured that was easier than extra code-gen options to disable it.

Reply:
You mean that you're not allowed to implement the service interface without embedding, because there's some private marker method now?

Reply:
Not because of a private market method -- it's enforced in the generated code differently than that.

The reason is because it has, from the beginning, been intended that adding a method to a service is a _backwards compatible_ change. But that isn't the case in older versions of the gRPC runtime: if you add a method to the service, then a new method gets added to the interface, and existing implementations could break. So the way to correct that breakage is to require this embedding, which ensures that if the interface gets new methods, these structs will still implement the interface.

There was a _ton_ of push-back on this by the community -- it's stinks to have code all compile and yet you have not actually updated all implementations of the service interface to handle the new method. For that camp (in which I would include FullStory), there is an option you can pass to `protoc-gen-go-grpc` to disable this requirement in the generated code.